crates/matrix-sdk-ui/src/timeline/tests/mod.rs

```diff
@@ -397,3 +398,24 @@ impl RoomDataProvider for TestRoomDataProvider {
         Some((push_rules, push_context))
     }
 }
+
+#[allow(unused_variables)]
```

Comment:
This warning fires for a good reason. In line 410, you're not actually comparing anything to the `index` parameter, you're just binding a local variable named `index` and discarding it immediately (by not using it in the `=>` match arm).